File: process_excel.py
import pandas as pd
from datetime import datetime
import os

def process_excel_to_csv():
    # Get today's date to find the correct file
    today = datetime.now()
    date_str = today.strftime("%d-%m-%Y")
    excel_filename = f"ventas_{date_str}.xlsx"
    csv_filename = "Negocio Bolsas.csv"

    # Check if today's Excel file exists
    if not os.path.exists(excel_filename):
        print(f"Error: File '{excel_filename}' not found. Please run the download script first.")
        return

    try:
        # First, let's see the available sheet names
        xls = pd.ExcelFile(excel_filename)
        print(f"Found sheet names: {xls.sheet_names}")

        # Read the specific sheet "Ventas" from the Excel file
        # We use header=None because the original script handles a headerless CSV
        df = pd.read_excel(excel_filename, sheet_name="Ventas", header=None)

        # Save the dataframe to a CSV file with UTF-8 encoding
        df.to_csv(csv_filename, index=False, header=False, encoding='utf-8-sig')

        print(f"Successfully converted the 'Ventas' sheet from '{excel_filename}' to '{csv_filename}'.")

    except Exception as e:
        print(f"An error occurred: {e}")

if __name__ == "__main__":
    process_excel_to_csv()

File: process_excel.js
import fs from 'fs';
import path from 'path';
import { pathToFileURL } from 'url';
import { processExcelToCsv } from './src/services/excel_processor_logic.js';
import {RUTA_VENTAS_CSV} from './src/config.js';

/**
 * Script para procesar un archivo Excel y convertirlo a CSV.
 * Esta versión mantiene compatibilidad con versiones anteriores pero
 * utiliza la lógica centralizada en excel_processor_logic.js.
 */
const processExcel = async () => {
  const today = new Date();
  const day = String(today.getDate()).padStart(2, '0');
  const month = String(today.getMonth() + 1).padStart(2, '0');
  const year = today.getFullYear();
  const dateStr = `${day}-${month}-${year}`;
  const excelFilename = `ventas_${dateStr}.xlsx`;
  const csvFilename = RUTA_VENTAS_CSV

  if (!fs.existsSync(excelFilename)) {
    console.error(`Error: File '${excelFilename}' not found. Please run the download script first.`);
    return;
  }

  try {
    // Usar la función centralizada de excel_processor_logic.js
    await processExcelToCsv(excelFilename, csvFilename, 'Ventas');

    console.log(`Successfully converted the 'Ventas' sheet from '${excelFilename}' to '${csvFilename}'.`);
  } catch (error) {
    console.error(`An error occurred: ${error}`);
  }
};

if (import.meta.url === pathToFileURL(process.argv[1]).href) {
  processExcel();
}
